src/MiniCover.HitServices/MethodScope.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.IO;
+
+namespace MiniCover.HitServices
+{
+    public class MethodScope : IDisposable
+    {
+        private readonly HitContext _hitContext;
+        private readonly string _hitsPath;
+        private readonly bool _createdHitContext;
+
+        public MethodScope(
+            string hitsPath,
+            string assemblyName,
+            string className,
+            string methodName)
+        {
+            _hitContext = HitContext.Current;
+
+            if (_hitContext == null)
+            {
+                _hitContext = new HitContext(assemblyName, className, methodName);
+                _createdHitContext = true;
+                HitContext.Current = _hitContext;
+            }
+
+            _hitsPath = hitsPath;
+            _hitContext.IncrementRef();
+        }
+
+        public void Hit(int id)
+        {
+            _hitContext.RecordHit(id);
+        }
+
+        public void Dispose()
+        {
+            if (_hitContext.DecrementRef() == 0)
+                SaveHitContext();
+
+            if (_createdHitContext)
+                HitContext.Current = null;
+        }
+
+        private void SaveHitContext()
+        {
+            Directory.CreateDirectory(_hitsPath);
+
+            var fileName = Path.Combine(_hitsPath, $"{Guid.NewGuid()}.hits");
```

Comment:
This is still wrong. When it rewrites the HitContext to disk it will generate a new id and duplicate the hits in 2 files.

I will change it to make sure the same HitContext is always written to the same file. Maybe I can optimize as well to only write the diff the second time the hit context is saved.